packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts

```diff
@@ -44,19 +54,39 @@ ${Object.entries(nextStaticHeaders)
 	}
 }
 
+/**
+ * Builds the `_routes.json` file, which is used by Cloudflare to determine which routes should
+ * invoke the worker.
+ *
+ * Collects existing entries from the `_routes.json` file in the root of the project, if it exists.
+ *
+ * @param outputDir Output directory for the metadata files.
+ * @param opts Options for building metadata files.
+ */
 async function buildRoutes(outputDir: string, opts: BuildMetadataFilesOpts) {
 	const nextStaticPath = getNextStaticDirPath(opts);
 
+	const existingFile = await readJsonFile<RoutesJsonFile>('_routes.json');
+	if (existingFile && existingFile.version !== 1) {
+		cliError(
+			`Found _routes.json with version ${existingFile.version}, but only version 1 is supported.`,
+		);
+		process.exit(1);
+	}
+
+	const { include = [], exclude = [] } = existingFile ?? {};
+	const includeEntries = include.length > 0 ? include : ['/*'];
+	const excludeEntries = new Set([`${nextStaticPath}/*`, ...(exclude ?? [])]);
+
 	try {
 		await writeFile(
 			join(outputDir, '_routes.json'),
 			JSON.stringify({
 				version: 1,
 				description: `Built with @cloudflare/next-on-pages@${nextOnPagesVersion}.`,
-				include: ['/*'],
-				exclude: [`${nextStaticPath}/*`],
+				include: includeEntries,
+				exclude: [...excludeEntries],
```

Comment:
why the spreading? 😮 

```suggestion
				exclude: excludeEntries,
```